packages/jupyter-ai/src/components/chat.tsx

```diff
@@ -31,9 +31,20 @@ function ChatBody({ chatHandler, setChatView: chatViewHandler }: ChatBodyProps):
   const [replaceSelection, setReplaceSelection] = useState(false);
   const [input, setInput] = useState('');
   const [selection, replaceSelectionFn] = useSelectionContext();
+  const [sendWithShiftEnter, setSendWithShiftEnter] = useState(true);
+
+  // Load the config once, to determine Shift+Enter behavior
+  useEffect(() => {
+    async function getConfig() {
+      const config = await AiService.getConfig();
+      setSendWithShiftEnter(config.send_with_shift_enter ?? true);
+    }
+
+    getConfig();
+  }, []);
```

Comment:
Why do we need another effect hook here? This can be implemented in the current one that fetches the chat history and config on initial render.